### PR TITLE
protecting the flow when field is 0

### PR DESCRIPTION
**Pull request by alberto-sanchez**

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: HeavyFlavorAnalysis/Onia2MuMu/src/Onia2MuMuPAT.cc [hunk 1]
```diff
@@ -134,8 +134,15 @@ Onia2MuMuPAT::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 	TransientVertex myVertex = vtxFitter.vertex(t_tks);
 
 	CachingVertex<5> VtxForInvMass = vtxFitter.vertex( t_tks );
-	Measurement1D MassWErr = massCalculator.invariantMass( VtxForInvMass, muMasses );
-	
+
+        Measurement1D MassWErr(jpsi.M(),-9999.);
```
Comment (davidlange6) [inline HeavyFlavorAnalysis/Onia2MuMu/src/Onia2MuMuPAT.cc]: instead of try-catch - please just check the field value directly
@alberto-sanchez

**Comment by alberto-sanchez** [comment]
@davidlange6 , thanks for suggestion. I have modified the code.

File: HeavyFlavorAnalysis/Onia2MuMu/src/Onia2MuMuPAT.cc [hunk 2]
```diff
@@ -136,10 +138,10 @@ Onia2MuMuPAT::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
...
 
         Measurement1D MassWErr(jpsi.M(),-9999.);
-        try {
-	  MassWErr = massCalculator.invariantMass( VtxForInvMass, muMasses );
-        } catch (std::exception & err) {
-          std::cout << " Field is 0, invalidating vertex " << std::endl;
+        if ( field->nominalValue() > 0 ) {
+          MassWErr = massCalculator.invariantMass( VtxForInvMass, muMasses );
+        } else {
+          std::cout << " Field is " << field->nominalValue() << " kGauss, invalidating vertex combination" << std::endl;
```
Comment (davidlange6) [inline HeavyFlavorAnalysis/Onia2MuMu/src/Onia2MuMuPAT.cc]: sorry i missed the cout  which should be removed- I guess you don't want to print anything this module should work routinely with B=0 (even if we hope not to have any such data!)

@alberto-sanchez

**Comment by alberto-sanchez** [comment]
@davidlange6 , yes we do not expect to run on such data, however we want to prevent that. Thanks!.

**Pull request merged**